lockbox-ios/Presenter/ItemListPresenter.swift

```diff
@@ -208,13 +210,35 @@ class ItemListPresenter {
         self.itemListDisplayActionHandler.invoke(ItemListSortingAction.alphabetically)
         self.itemListDisplayActionHandler.invoke(ItemListFilterAction(filteringText: ""))
 
-        guard let view = self.view,
-              let sortButtonObserver = view.sortingButtonEnabled else {
-            return
+        guard let view = self.view else { return }
+
+        let syncingObserver = self.dataStore.syncState.map {(syncState: SyncState) -> Bool in
+            return syncState == .Syncing
         }
 
+        let isManualRefreshObserver = self.itemListDisplayStore.listDisplay
+                .filterByType(class: PullToRefreshAction.self)
+
+        Observable.combineLatest(syncingObserver, isManualRefreshObserver)
+            .map { (latest: (Bool, PullToRefreshAction)) -> Bool in
+                let isSyncing = latest.0
+                let isManualSync = latest.1.refreshing
+                return isSyncing && isManualSync
+        }.bind(to: view.pullToRefreshActive)
+            .disposed(by: self.disposeBag)
+
+        self.dataStore.syncState.filter { (syncState: SyncState) -> Bool in
+                return syncState == .Synced
+            }.subscribe(onNext: { (_) in
+                self.itemListDisplayActionHandler.invoke(PullToRefreshAction(refreshing: false))
```

Comment:
This could be done in the `do(onNext:{})` operator for the previous call. You've already set up the `syncingObserver`, so even if you were going to keep this separation (I think there's an argument for separating that kind of logic tbh) I would use `syncingObserver` rather than re-implementing the filter.

Reply:
This one is checking for the `Synced` state in order to dismiss the pull to action. `syncingObserver` is for when it is engaged in the syncing process.

Reply:
...oops, you're right !!! I can't read ;)